api/product.rb

```diff
@@ -101,7 +104,17 @@ def version_obj(name)
     end
 
     def version_obj_or_default(name)
```

Comment:
version_obj feels.. ok.. since it's really just wrapping the version class around a version string. It's also going to be a bit complex to rip that particular method out.